### DISCUSSION TITLE
Moving to Deno

### DISCUSSION THREAD

**Discussion by K0IN** [K0IN/Notify#74] [Announcements]
The project will be ported to deno, due to various annoyances with the js/npm/node ecosystem.

In the last time, I had so many weird bugs due to miniflare/node, that I simply call it quit. 
I will rework this project to use Deno (https://deno.land/) instead, I hope this makes it easier to host (deno deploy/docket/native).

This might take some time, I will move this code onto a legacy branch.

**Comment by K0IN** [comment]
I think im ready to release the new deno version.
THIS WILL INTRODUCE BREAKING CHANGES!
If you rely on our interface, please use the legacy branch to build your instance from source.
I will merge the changes on Sunday, the 30.07.2023 :)

**Comment by real-timer** [comment]
Can we have a version that follows the https://developer.mozilla.org/en-US/docs/Web/API/Notification/Notification API verbatim?
It should take an array of arguments like `[title, options]` and options should follow the web standard APIs. Forget about the weird, arbitrary shape in https://github.com/K0IN/Notify/blob/main/doc/api.md.

**Reply by K0IN** [reply]
The shape was chosen for simplicity.
We can add this to the 2.0, some time in the future :) (there are way more important things to-do)

**Reply by K0IN** [reply]
Notifications are shown here: https://github.com/K0IN/Notify/blob/61649f2d64561991c4a00c5ece0e2a298519a0c5/app/frontend/src/sw.js#L84C27-L84C43}

Since we are in a service worker context.
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification

I will keep it in mind.
